### PR TITLE
Add partition field for catalog deregister docs

### PR DESCRIPTION
**Pull request by jm96441n**
### Description

We accept a Partition field for deregistering nodes and it is required if you're trying to deregister a node that only exists in a particular partition.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: website/content/api-docs/catalog.mdx [hunk 1]
```diff
@@ -212,6 +212,8 @@ The behavior of the endpoint depends on what keys are provided.
   You can also specify the namespace in the `Service` or `Check` fields;
   if namespaces are specified in multiple places, they must all be the same.
 
+- `Partition` `(string: "")` <EnterpriseAlert inline /> - Specifies the partition of the service and checks you deregister.
```
[the remark this answers is not in the document]
Comment (jm96441n) [reply website/content/api-docs/catalog.mdx]: good call! thanks for reviewing this so quickly

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @jm96441n! a backport is missing for this PR [21788] for versions [1.15,1.17,1.18,1.19] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Pull request merged**